### ISSUE TITLE
error with hattip when using vite-plugin-graphql-codegen

### ISSUE THREAD

**Issue by sharmapukar217** [rakkasjs/rakkasjs#50]
TypeError: response.body[Symbol.asyncIterator] is not a function at Module.default (file://<directory>/node_modules/@hattip/adapter-node/dist/chunk-SPL6Q7DV.js:56:61) at async file://<directory>/node_modules/@vavite/connect/dist/index.mjs:66:15

**Comment by cyco130** [comment]
Can you provide a small reproduction please?

**Comment by sharmapukar217** [comment]
uncomment line numbers 6 and 25 in `vite.config.ts` to get the issue I'm facing 😢

**Comment by sharmapukar217** [comment]
also, vite-plugin-mkcert doesn't seems to work with hattip. To reproduce the error for https error, just run `yarn dev --https`

**Comment by cyco130** [comment]
Hi, the first issue is caused by `vite-plugin-graphql-codegen` installing a `fetch` polyfill that is not compatible with HatTip. One possible workaround is to make sure that HatTip's fetch polyfill is installed before codegen's:

```ts
// vite.config.ts
import install from "@hattip/polyfills/node-fetch";
install();
const codegen = await import("vite-plugin-graphql-codegen").then(
  (m) => m.default
);
```

But the easiest fix is to **upgrade to Node >= 18.6.0** if possible. It supports fetch natively.

The HTTPS issue is indeed a HatTip bug. I've fixed it and I will soon release a new HatTip version along with a new Rakkas version that uses the newer HatTip version. In the meantime, **upgrading to Node >= 18.6.0** solves this too.
